### PR TITLE
Draft: WIP: Fixed login graceful exit bug

### LINKED ISSUE #31: tabcmd2 login
When a user enters `tabmcd2 login`without giving the user credentials via flags, call stack of errors is shown. This error must be handled by asking the user to check for parameters and exit gracefully.

### PR DESCRIPTION
**Pull request by ovinis**
Fixes #31

### CODE REVIEW

File: tabcmd/commands/auth/session.py [hunk 1]
```diff
@@ -145,31 +153,40 @@ def no_cookie_save_session_creation_with_token(self):
...
+                                 "session========")
+                signed_in_object = self.reuse_session()
+            if args.no_cookie:
+                self.remove_json()
+            else:
+                self.save_token_to_json_file()
+            return signed_in_object
+        except (Exception,):
+            self.logger.error("Cannot create a session, Please try "
+                              "again with updated credentials")
```
Comment (jacalata) [inline tabcmd/commands/auth/session.py:188]: You could extract this to a standalone login_failed method

File: tabcmd/parsers/login_parser.py [hunk 2]
```diff
@@ -21,7 +21,8 @@ def login_parser():
         subparsers = parser.add_subparsers()
         login_parser = subparsers.add_parser('login', parents=[parser])
         args = login_parser.parse_args(sys.argv[LoginParser.USER_ARG_IDX:])
-
+        print(args)
+        print("PROMPTS: ", args.prompt)
```
Comment (jacalata) [inline tabcmd/parsers/login_parser.py:25]: These seem like debug printlines? better to use the logger so they don't get accidentally left in :)

**Review by jacalata — CHANGES_REQUESTED** [review]

File: tabcmd/commands/auth/session.py [hunk 3]
```diff
@@ -113,15 +114,22 @@ def no_cookie_save_session_creation_with_username(self):
...
-        tableau_server._auth_token = self.auth_token
-        tableau_server._site_id = self.site_id
-        return tableau_server
+        try:
+            tableau_server = TSC.Server(self.server,
+                                        use_server_version=True)
+            tableau_server._auth_token = self.auth_token
+            tableau_server._site_id = self.site_id
+            return tableau_server
+        except TSC.ServerResponseError as e:
```
Comment (mmuttreja-tableau) [inline tabcmd/commands/auth/session.py:128]: Why are we only printing one error? can we have a default except error 


```tabcmd2 login
INFO pythontabcmd2.login_command  ========Creating a new session========
INFO pythontabcmd2.session  ==========Continuing previous session========
Traceback (most recent call last):
  File "/usr/local/bin/tabcmd2", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/pythontabcmd2/tabcmd2.py", line 9, in main
    command_context.execute_command()
  File "/usr/local/lib/python3.9/site-packages/pythontabcmd2/context.py", line 34, in execute_command
    command_strategy.run_command()
  File "/usr/local/lib/python3.9/site-packages/pythontabcmd2/commands/auth/login_command.py", line 22, in run_command
    self.create_session_login_command(self.args)
  File "/usr/local/lib/python3.9/site-packages/pythontabcmd2/commands/auth/login_command.py", line 28, in create_session_login_command
    session.create_session(args)
  File "/usr/local/lib/python3.9/site-packages/pythontabcmd2/commands/auth/session.py", line 167, in create_session
    signed_in_object = self.reuse_session()
  File "/usr/local/lib/python3.9/site-packages/pythontabcmd2/commands/auth/session.py", line 120, in reuse_session
    tableau_server = TSC.Server(self.server,
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/server.py", line 92, in __init__
    self.use_server_version()
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/server.py", line 132, in use_server_version
    self.version = self._determine_highest_version()
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/server.py", line 122, in _determine_highest_version
    version = self.server_info.get().rest_api_version
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 177, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/endpoint/server_info_endpoint.py", line 22, in get
    server_response = self.get_unauthenticated_request(self.baseurl)
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 100, in get_unauthenticated_request
    return self._make_request(self.parent_srv.session.get, url)
  File "/usr/local/lib/python3.9/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 69, in _make_request
    server_response = method(url, **parameters)
  File "/usr/local/lib/python3.9/site-packages/requests-2.25.1-py3.9.egg/requests/sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests-2.25.1-py3.9.egg/requests/sessions.py", line 528, in request
    prep = self.prepare_request(req)
  File "/usr/local/lib/python3.9/site-packages/requests-2.25.1-py3.9.egg/requests/sessions.py", line 456, in prepare_request
    p.prepare(
  File "/usr/local/lib/python3.9/site-packages/requests-2.25.1-py3.9.egg/requests/models.py", line 316, in prepare
    self.prepare_url(url, params)
  File "/usr/local/lib/python3.9/site-packages/requests-2.25.1-py3.9.egg/requests/models.py", line 390, in prepare_url
    raise MissingSchema(error)
requests.exceptions.MissingSchema: Invalid URL 'None/api/2.4/serverInfo': No schema supplied. Perhaps you meant http://None/api/2.4/serverInfo?
```

**Comment by ovinis** [comment]
Closing this PR as Jac's new Refactoring PR (#56) fixes the login related bugs discussed here

**Pull request closed without merging**